File: packages/case-definition-dsl/src/matchers/arrays/ArrayStartsWith.ts
import { SHAPED_ARRAY_MATCHER_TYPE } from '@contract-case/case-entities-internal';
import { AnyMatcher } from '../base';
import { AnyMatcherOrData } from '../../types';

/**
 * Matches an Array which starts with the provided array of matchers - any additional elements in the array are ignored.
 * @public
 */
export class ArrayStartsWith extends AnyMatcher {
  /** @internal */
  readonly '_case:matcher:type': typeof SHAPED_ARRAY_MATCHER_TYPE;

  /** @internal */
  readonly '_case:matcher:children': Array<AnyMatcherOrData>;

  /**
   * Matches an Array which starts with the provided array of matchers - any additional elements in the array are ignored.
   *
   * @param matchers -  An array of matchers that describes the start of the array. Additional elements in the actual array are ignored.
   */
  constructor(matchers: Array<AnyMatcherOrData>) {
    super(SHAPED_ARRAY_MATCHER_TYPE);
    this['_case:matcher:children'] = matchers;
  }

  /**
   * For non-TypeScript implementations (see `AnyMatcher.toJSON`)
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/arrays/ArrayStartsWith.java
package io.contract_testing.contractcase.dsl.matchers.arrays;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import jakarta.annotation.Generated;
import java.lang.Object;
import java.lang.String;
import java.util.List;
import javax.annotation.Nullable;
import javax.annotation.Nullable;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;

/**
 * Matches an Array which starts with the provided array of matchers - any additional elements in the array are ignored.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class ArrayStartsWith implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * An array of matchers that describes the start of the array. Additional elements in the actual array are ignored.
   */
  @Nullable
  @JsonInclude(Include.NON_NULL)
  @Getter
  @JsonProperty("_case:matcher:children")
  private final List<Object> children;

  @Builder
  public ArrayStartsWith(@Nullable final List<Object> children) {
    this.type = "_case:ArrayShape";
    this.children = children;
  }
}
